aleph/tests/test_role_model.py

```diff
@@ -98,3 +100,33 @@ def test_role_by_api_key_empty(self):
 
         role = Role.by_api_key("")
         assert role is None
+
+    def test_role_by_api_key_expired(self):
+        role_ = self.create_user()
+        role_.api_key = "1234567890"
+        role_.api_key_expires_at = datetime.datetime(2024, 3, 31, 0, 0, 0)
+        db.session.add(role_)
+        db.session.commit()
+
+        with time_machine.travel("2024-03-30T23:59:59Z"):
+            print(role_.api_key_expires_at)
+            role = Role.by_api_key(role_.api_key)
+            assert role is not None
+            assert role.id == role_.id
+
+        with time_machine.travel("2024-03-31T00:00:00Z"):
+            role = Role.by_api_key(role_.api_key)
+            assert role is None
+
+    def test_role_by_api_key_legacy_without_expiration(self):
```

Comment:
Will this test have an expiry date after which it wouldn't make sense to keep it any longer?

Reply:
Yes, although it should probably not be replaced at that point and instead simply be updated to ensure that any keys that do not have an expiration date are always invalid or something like that. I’ll add a comment to the code and open an issue to keep track of this.